nautobot_circuit_maintenance/handle_notifications/handler.py

```diff
@@ -164,6 +164,22 @@ def create_or_update_circuit_maintenance(
     maintenance_id = f"{raw_entry.provider.slug}-{parser_maintenance.maintenance_id}"
     try:
         circuit_maintenance_entry = CircuitMaintenance.objects.get(name=maintenance_id)
+        last_parsed_notification = (
+            circuit_maintenance_entry.parsednotification_set.order_by("raw_notification__date").reverse().last()
+        )
+
+        # If the notification is older than the latest one used to update the CircuitMaintenance, we skip updating it
+        parser_maintenance_datetime = datetime.datetime.fromtimestamp(
+            parser_maintenance.stamp, tz=datetime.timezone.utc
+        )
+        if last_parsed_notification and last_parsed_notification.date > parser_maintenance_datetime:
+            if logger.debug:
```

Comment:
I think I'd lean towards always logging this message, rather than gating it on the `debug` flag.